barretenberg/cpp/src/barretenberg/goblin/full_goblin_composer.test.cpp

```diff
@@ -145,52 +45,31 @@ class FullGoblinComposerTests : public ::testing::Test {
  */
 TEST_F(FullGoblinComposerTests, SimpleCircuit)
 {
-    auto op_queue = std::make_shared<proof_system::ECCOpQueue>();
+    barretenberg::Goblin goblin;
 
-    // Add mock data to op queue to simulate interaction with a "first" circuit
-    perform_op_queue_interactions_for_mock_first_circuit(op_queue);
+    // Construct an initial circuit; its proof will be recursively verified by the first kernel
+    info("Initial circuit.");
+    GoblinUltraBuilder initial_circuit{ goblin.op_queue };
+    GoblinTestingUtils::construct_simple_initial_circuit(initial_circuit);
+    KernelInput kernel_input = goblin.accumulate(initial_circuit);
 
     // Construct a series of simple Goblin circuits; generate and verify their proofs
-    size_t NUM_CIRCUITS = 3;
+    size_t NUM_CIRCUITS = 2;
     for (size_t circuit_idx = 0; circuit_idx < NUM_CIRCUITS; ++circuit_idx) {
-        proof_system::GoblinUltraCircuitBuilder builder{ op_queue };
-
-        generate_test_circuit(builder);
 
-        // The same composer is used to manage Honk and Merge prover/verifier
-        proof_system::honk::GoblinUltraComposer composer;
+        // Construct a circuit with logic resembling that of the "kernel circuit"
```

Comment:
Its a bit misleading to use the kernel terminology here since the point of this test is that there's no recursion. Not a big deal